File: adminportal/src/ApiResponses/ApiFileResponse.ts
import { ApiResponse } from './ApiResponse';

export interface IApiFileResponse {
    responseMessage: string | null,
    fileContents: string,
    contentType: string,
    fileDownloadName: string,
    fileName: string
}

export class ApiFileResponse extends ApiResponse<string> implements IApiFileResponse {
    constructor(
        public responseMessage: string | null,
        public fileContents: string,
        public contentType: string,
        public fileDownloadName: string,
        public fileName: string
    ) {
        super(true, responseMessage || 'File retrieved successfully', fileContents);
    }
}

File: server/RestApiServer/Core/ApiResponses/ApiFileResponse.cs
using Microsoft.Net.Http.Headers;

namespace RestApiServer.Core.ApiResponses
{
    public class ApiFileResponse
    {
        public string? ResponseMessage { get; set; }
        public byte[] FileContents { get; set; } = [];
        public string ContentType { get; set; } = "";
        public string FileDownloadName { get; set; } = "";
        public string FileName { get; set; } = "";
    }
}